client/web/src/components/login-toggle.tsx

```diff
@@ -134,50 +134,89 @@ function LoginPopoverContent({
     }
   }, [node.IP, auth.viewerIdentity, newSession])
 
+  const signInButton = useCallback(() => {
+    return (
+      <button
+        className={cx(
+          "w-full px-3 py-2 bg-indigo-500 rounded shadow text-center text-white text-sm font-medium mt-2",
+          {
+            "mb-2": auth.viewerIdentity,
+          }
+        )}
+        onClick={handleSignInClick}
+      >
+        {auth.viewerIdentity ? "Sign in to confirm identity" : "Sign in"}
+      </button>
+    )
+  }, [auth, handleSignInClick])
+
   return (
     <div onMouseEnter={!canConnectOverTS ? checkTSConnection : undefined}>
       <div className="text-black text-sm font-medium leading-tight mb-1">
         {!auth.canManageNode ? "Viewing" : "Managing"}
         {auth.viewerIdentity && ` as ${auth.viewerIdentity.loginName}`}
       </div>
-      {!auth.canManageNode &&
-        (!auth.viewerIdentity || auth.authNeeded === AuthType.tailscale ? (
```

Comment:
if they don't have permission to manage the node, then that means we know who they are.  So this check `!auth.viewerIdentity` would fail, then `auth.authNeeded === AuthType.tailscale` should fail I'm pretty sure, since we're not going to return that if we know the user doesn't have access, so then they would fall through to the final else at line 212 (which is still inside the `!auth.canManageNode` block).  So I think this works?

Reply:
Ugh yeah I was getting confused by the ternaries. 😵  Thought 194 would get evaluated. Okay this look like a safe change to me then.

Reply:
is there a cleaner way to do this in react without all the ternaries?